packages/venia-concept/src/components/Head/headProvider.js

```diff
@@ -0,0 +1,19 @@
+import React, { useEffect } from 'react';
+import { HeadProvider as _HeadProvider } from 'react-head';
+
+const filterOutTitleTags = titleTags => {
+    titleTags
+        .filter(titleNode => !titleNode.dataset.synteticTag)
+        .forEach(titleTag => {
+            if (titleTag) titleTag.parentNode.removeChild(titleTag);
+        });
+};
+
+export default props => {
```

Comment:
For consistency in the codebase let's give this function a semantic name and export it separately.

```js
const HeadProvider = props => { ... }

export default HeadProvider;
```